src/Interpreters/executeQuery.cpp

```diff
@@ -108,12 +108,18 @@ static void checkASTSizeLimits(const IAST & ast, const Settings & settings)
 }
 
 
-static String prepareQueryForLogging(const String & query, ContextPtr context)
+static String prepareQueryForLogging(const String & query, const ASTPtr & ast, ContextPtr context)
 {
     String res = query;
 
-    // Wiping a password or its hash from CREATE/ALTER USER query because we don't want it to go to logs.
-    res = wipePasswordFromQuery(res);
+    // Wiping a password or hash from CREATE/ALTER USER query because we don't want it to go to logs.
+    if (ast)
+    {
+        ASTPtr ast_for_logging = ast;
+        wipePasswordFromQuery(ast_for_logging);
```

Comment:
Maybe check `ast->as<ASTCreateUserQuery>` here and call wipePasswordFromQuery returning `ASTPtr`? it's minor and a matter of taste. For me, checking `ast_for_logging != ast` looks confusing and requires reading the code of `wipePasswordFromQuery` to know that it can clone a query and how it's modified.

Reply:
I've made this part more clear for undestanding by adding new function `canContainPassword()`.

I don't want to check `ast->as<ASTCreateUserQuery>` here in `executeQuery.cpp` because I think I'll add more cases for wiping passwords in future PRs. Because it seems it's necessary to remove passwords not only from `ASTCreateUserQuery` but also from some other types of queries.